src/main/java/com/lirc572/ip/Parser.java

```diff
@@ -0,0 +1,218 @@
+package com.lirc572.ip;
+
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Contains static methods for parsing commands.
+ */
+public class Parser {
+
+    /**
+     * All exit commands are listed here.
+     */
+    private static final Set<String> EXIT_COMMANDS = Set.of(
+            "bye", "exit", "quit"
+    );
+
+    /**
+     * Tokenizes a command.
+     *
+     * @param command A line of command.
+     * @return The tokenized command as a String array.
+     */
+    public static String[] tokenizeCommand(String command) {
+        ArrayList<String> tokens = new ArrayList<>();
+        boolean newToken = true;
+        boolean unmatchedQuote = false;
+        char quotationMark = ' ';
```

Comment:
Consider using isNewToken and isUnmatchedQuote to describe your boolean variables.